components/dashboard/src/workspaces/WorkspaceEntry.tsx

```diff
@@ -74,7 +74,9 @@ export const WorkspaceEntry: FunctionComponent<Props> = ({ info }) => {
                 </a>
             </ItemField>
             <ItemField className="w-2/12 flex flex-col my-auto">
-                <div className="text-gray-500 dark:text-gray-400 overflow-ellipsis truncate">{currentBranch}</div>
+                <div className="text-gray-500 dark:text-gray-400 overflow-ellipsis truncate">
+                    <Tooltip content={currentBranch}>{currentBranch}</Tooltip>
```

Comment:
Some raw thoughts on this change:
1. It's better to avoid as many tooltips as possible, mostly for accessibility.
2. Tooltip seems redundant when the branch is not truncated. 
3. Tooltip seems redundant when the default branch is by convention short like `master` or `main`.
4. This could use the [`title`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title) attribute as a fallback instead.
5. The tooltip will become less needed with changes in https://github.com/gitpod-io/gitpod/issues/12033, see also screenshot below.
6. Following up from the changes in https://github.com/gitpod-io/gitpod/issues/12033 the branch could be truncated in some edge cases but still be accompanied by a copy-to-clipboard button next to the branch name.

| Early design for improving the workspaces list |
|-|
| <img width="1456" alt="Workspaces" src="https://user-images.githubusercontent.com/120486/218989801-d1f9996c-ccdb-42fb-b3da-0b665c9fbc49.png"> |

Reply:
Yah the new design will help so we wouldn't need something like this. My motivation is that I end up with the following situation:

![image](https://user-images.githubusercontent.com/367275/219061994-2c3cda44-c577-464d-bf80-b54d60815c8b.png)

and the only way I can tell what's what is by closing any sidebars, and hoping it doesn't clip (and still does sometimes). This seemed like a reasonable / easy way to offer a recourse for a user to see their branch.

I'm not sure I follow what accessibility concerns there would be with this. Using our custom tooltip component is essentially the same behavior as using title, but a slightly better ux.

I do agree it is redundant, and shouldn't be needed (and won't in new designs whenever we can do them). At the moment, I was hoping to improve it slightly until we do have time to work on the new designs. It's easy enough to remove, so I will if you still think we should, but also feels like a quality of life improvement that I know I'd benefit from, so I assume some others will to.

Reply:
@selfcontained I'm coming to this a bit late to share some thoughts on this topic—I also understand where you are coming from and I'm looking forward to improving the bits in this page.

> I'm not sure I follow what accessibility concerns there would be with this. Using our custom tooltip component is essentially the same behavior as using title, but a slightly better ux.

To clarify, I believe there are cases where tooltips can be a helpful and valuable component for surfacing information like describing a button action, provide a hint, etc. However, it's easy to skip making it accessible, as **tooltips that only appear on mouse over are inaccessible for users that rely on keyboards to navigate**. This is a perfect example as the branch element is not accessible by keyboard and the tooltip never shows up.

> I do agree it is redundant, and shouldn't be needed (and won't in new designs whenever we can do them).

💯 

> ... I was hoping to improve it slightly until we do have time to work on the new designs. It's easy enough to remove, so I will if you still think we should, but also feels like a quality of life improvement that I know I'd benefit from, so I assume some others will to.

Sounds like a good [MVC](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e?pvs=4#0c2e0d6c8a99404b97ccdc206b573d72) we can easily update or revert when we start working on the new page designs. Also, since the changes in https://github.com/gitpod-io/gitpod/pull/15716 this seems like ok to leave it there for now. 🤝 

---

Also, regarding the comment below in https://github.com/gitpod-io/gitpod/pull/16397#issuecomment-1431749162:
> I noticed something with tooltip change I mad here makes it hard to hover over the tip and select/copy the value now.

Since a tooltip disappears when a user hovers away, we should not include information that is pertinent for the user to complete a task. We should instead rely on other patterns or components for making vital information always visible and accessible, like help text, lables, etc.